### DISCUSSION TITLE
Middle mouse button functionality in costume editor

### DISCUSSION THREAD

**Discussion by Illodrus** [ScratchAddons/ScratchAddons#1888] [Ideas]
If one of these would be possible to implement, it would be very helpful:

1. Auto scroll in costume editor after clicking middle mouse button, click again to disable
2. Move viewport in costume editor by holding down middle mouse button and dragging

**Comment by Mr-Programmer-777** [comment]
I like it! Would be very useful. 😄

**Comment by Exlectruz** [comment]
Sounds helpful!

**Comment by Explosion-Scratch** [comment]
I think that the middle mouse button is unintuitive. I think it'd be better as a corner click option.

**Reply by Illodrus** [reply]
It's based on a Google Chrome feature and a feature in a lot of other editors/engines that allows you to hold the middle mouse to drag or, in the case of Chrome, click the middle mouse once to have auto-scroll. Even if it is a little unintuitive, from my experience it's very helpful, especially at a high zoom level
